src/views/CanvasOrderbook/OrderbookControls.tsx

```diff
@@ -0,0 +1,143 @@
+import { useCallback } from 'react';
+
+import { clamp } from 'lodash';
+import { shallowEqual } from 'react-redux';
+import styled from 'styled-components';
+
+import { MarketOrderbookGrouping, Nullable, OrderbookGrouping } from '@/constants/abacus';
+import { ButtonShape, ButtonSize } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+
+import { Button } from '@/components/Button';
+import { Output, OutputType } from '@/components/Output';
+import { ToggleGroup } from '@/components/ToggleGroup';
+
+import { useAppSelector } from '@/state/appTypes';
+import { getCurrentMarketConfig } from '@/state/perpetualsSelectors';
+
+import abacusStateManager from '@/lib/abacus';
+
+type OrderbookControlsProps = {
+  className?: string;
+  assetName?: string;
+  selectedUnit: 'fiat' | 'asset';
+  setSelectedUnit(val: 'fiat' | 'asset'): void;
+  grouping: Nullable<MarketOrderbookGrouping>;
+};
+
+export const OrderbookControls = ({
+  className,
+  assetName,
+  selectedUnit,
+  setSelectedUnit,
+  grouping,
+}: OrderbookControlsProps) => {
+  const stringGetter = useStringGetter();
+  const modifyScale = useCallback(
+    (direction: number) => {
+      const start = grouping?.multiplier.ordinal ?? 0;
+      const end = clamp(start + direction, 0, 3);
+      abacusStateManager.modifyOrderbookLevel(
+        OrderbookGrouping.values().find((v) => v.ordinal === end)!
+      );
+    },
+    [grouping?.multiplier.ordinal]
+  );
+  const currentMarketConfig = useAppSelector(getCurrentMarketConfig, shallowEqual);
+  const { tickSizeDecimals: tSD } = currentMarketConfig ?? {};
+  const tickSizeDecimals = tSD ?? 2;
```

Comment:
instead of destructuring
`const tickSizeDecimals = currentMarketConfig?.tickSizeDecimals ?? USD_DECIMALS`